From 697c36441e000644f63bb3952ebff0ccd52e1403 Mon Sep 17 00:00:00 2001
From: Christian Bruel <christian.bruel@foss.st.com>
Date: Mon, 5 Jun 2023 13:25:16 +0200
Subject: [PATCH] dt-bindings: PCI: Add STM32MP25 PCIe EP bindings

STM32MP25 PCIe Controller is based on the DesignWare core configured as
end point mode from the SYSCFG register.

Signed-off-by: Christian Bruel <christian.bruel@foss.st.com>
Change-Id: Ia67cfd41261300b78b9ae8c1cf8118f305e98821
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/321964
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 .../bindings/pci/st,stm32-pcie-ep.yaml        | 101 ++++++++++++++++++
 1 file changed, 101 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/pci/st,stm32-pcie-ep.yaml

--- /dev/null
+++ b/Documentation/devicetree/bindings/pci/st,stm32-pcie-ep.yaml
@@ -0,0 +1,101 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/pci/st,stm32-pcie-ep.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: PCIe EP controller on STM32MP25 SoC
+
+maintainers:
+  - Christian Bruel <christian.bruel@foss.st.com>
+
+description: |
+  PCIe EP controller based on the Synopsys DesignWare PCIe core.
+
+allOf:
+  - $ref: /schemas/pci/snps,dw-pcie-ep.yaml#
+
+properties:
+  compatible:
+    const: st,stm32mp25-pcie-ep
+
+  reg:
+    items:
+      - description: Data Bus Interface (DBI) registers.
+      - description: PCIe configuration registers.
+
+  reg-names:
+    items:
+      - const: dbi
+      - const: addr_space
+
+  st,syscfg:
+    $ref: /schemas/types.yaml#/definitions/phandle-array
+    description: Phandle to the SYSCON entry required for configuring PCIe mode.
+
+  clocks:
+    maxItems: 1
+    description: PCIe system clock
+
+  clock-names:
+    const: core
+
+  resets:
+    maxItems: 1
+
+  reset-names:
+    const: pcie
+
+  phys:
+    maxItems: 1
+
+  phy-names:
+    const: pcie-phy
+
+  reset-gpios:
+    description: GPIO controlled connection to PERST# signal
+    maxItems: 1
+
+  power-domains:
+    maxItems: 1
+
+  access-controllers:
+    maxItems: 1
+
+required:
+  - st,syscfg
+  - resets
+  - reset-names
+  - clocks
+  - clock-names
+  - phys
+  - phy-names
+  - reset-gpios
+
+unevaluatedProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/phy/phy.h>
+    #include <dt-bindings/clock/st,stm32mp25-rcc.h>
+    #include <dt-bindings/gpio/gpio.h>
+    #include <dt-bindings/reset/st,stm32mp25-rcc.h>
+
+    pcie_ep: pcie-ep@48400000 {
+             compatible = "st,stm32mp25-pcie-ep";
+             num-lanes = <1>;
+             reg = <0x48400000 0x400000>,
+                   <0x10000000 0x8000000>;
+             reg-names = "dbi", "addr_space";
+             st,syscfg = <&syscfg>;
+             clocks = <&rcc CK_BUS_PCIE>;
+             clock-names = "core";
+             phys = <&combophy PHY_TYPE_PCIE>;
+             phy-names = "pcie-phy";
+             resets = <&rcc PCIE_R>;
+             reset-names = "pcie";
+             pinctrl-names = "default", "init";
+             pinctrl-0 = <&pcie_pins_a>;
+             pinctrl-1 = <&pcie_init_pins_a>;
+             reset-gpios = <&gpioj 8 GPIO_ACTIVE_LOW>;
+    };
